api/envoy/config/filter/http/dynamic_forward_proxy/v2alpha/dynamic_forward_proxy.proto

```diff
@@ -24,8 +24,15 @@ message FilterConfig {
 
 // Per route Configuration for the dynamic forward proxy HTTP filter.
 message PerRouteConfig {
-  // Indicates that before DNS lookup, the host header will be swapped with
-  // this value. If not set or empty, the original host header value
-  // will be used and no rewrite will happen.
-  string host_rewrite = 1;
+  oneof host_rewrite_specifier {
+    // Indicates that before DNS lookup, the host header will be swapped with
+    // this value. If not set or empty, the original host header value
+    // will be used and no rewrite will happen.
+    string host_rewrite = 1;
```

Comment:
Maybe `host_rewrite_literal` and `host_rewrite_header` or something to make it clear the distinction?

Also, what is the implication of the host rewrite, is this something that occurs for both the DNS lookup and host header forwarding?

One other consideration; how does this interact with HCM host rewriting?

Reply:
Yeah, we can go with `host_rewrite_literal` and `host_rewrite_header` -- I did however wanted to stay consistent with the HCM related configs... But maybe better to make them different.

Yeah, this is both for DNS lookup (as the comment says) and for host header forwarding. I can add comment specifying for the latter occurs as well.

This probably shouldn't be used with HCM host rewriting; if it is the value set here would be used for DNS lookups whereas the value in HCM will be use for host header forwarding... an unlikely desired outcome, I think.

Reply:
If you could document these interactions as part of the PR it would be great, thanks.

Reply:
@htuch hmm on a 2nd thought, how about we document how the HCM settings and the dynamic fwd proxy ones interact (my last commit does this) _but_ we leave the properties with the same current names as we have in the HCM? Thus, we don't pay the cognitive cost of new names, but we clearly document how they behave differently in this context.

Reply:
@rgs1 sounds good, but can you leave a `[#next-major-version` annotation here and in HCM to allow us to clean up both next time? Thanks

Reply:
done (used the new `[#rename-at-next-major-version: host_rewrite_literal]` tag btw).

Reply:
Ack, That bit looks good, resolving.